### DISCUSSION TITLE
Github repo dependency versioned with git tags gets "skipReason": "unknown-version"

### DISCUSSION THREAD

**Discussion by Simon-HL** [renovatebot/renovate#12841] [Request Help] [answered]
I am trying to use Renovate to lookup outdated dependencies in the form of github repos that are versioned as tags. 
Renovate doesn't seem to understand this versioning and skips my repo and logs "skipReason": "unknown-version".
What am I doing wrong here? 

This is my dependencies in package.json:
```
"dependencies": {
    "@simon-hl/testing-renovate-private-dependency-npm": "1.1.0",
    "dayjs": "1.10.7",
    "testing-renovate-private-dependency": "github:Simon-HL/Testing-Renovation-PrivateDependency#2.1.1"   <----- my github repo
  }
```
Here's a snippet of the renovate log:
```
{
    "depType": "dependencies",
    "depName": "testing-renovate-private-dependency",
    "currentValue": "github:Simon-HL/Testing-Renovation-PrivateDependency#2.1.1",
    "skipReason": "unknown-version",
    "prettyDepType": "dependency",
    "lockedVersion": null,
    "depIndex": 2,
    "updates": []
  }
],
```

The github repo used as dependency contains tags with a mix of npm versioning and git tags. 
E.g. Tags: 2.1.3, 2.1.2, v2.1.2, 2.1.1

I have tried adding a packagerule to explicitly state versioning, but to no success 

```
"packageRules": [
    {
      "matchDatasources": ["git-tag"],
      "matchPackageNames": ["testing-renovate-private-dependency"],
      "versioning": "git"
    }
  ]
```

**Comment by rarkins** [comment, marked as answer]
At first glance, it looks like our github validation regex is looking for only lower case: https://github.com/renovatebot/renovate/blob/62936413261a54829131f3e6354d7ffc8a8db2d0/lib/manager/npm/extract/index.ts#L298

However it could be deeper than that. Can you try lowering the case of the string and see if it works? If so then we should fix the validation to allow upper too.

**Reply by Simon-HL** [reply]
Thanks for the prompt reply.
You are right. After changing to lower case Renovate was able to understand the versioning. 

However, now it fails to lookup the dependency. 
```
{
  "depType": "dependencies",
  "depName": "testing-renovate-private-dependency",
  "currentValue": "2.1.1",
  "currentRawValue": "github:simon-hl/testing-renovation-privatedependency#2.1.1",
  "datasource": "github-tags",
  "lookupName": "simon-hl/testing-renovation-privatedependency",
  "pinDigests": false,
  "githubRepo": "simon-hl/testing-renovation-privatedependency",
  "sourceUrl": "https://github.com/simon-hl/testing-renovation-privatedependency",
  "gitRef": true,
  "prettyDepType": "dependency",
  "lockedVersion": null,
  "depIndex": 2,
  "updates": [],
  "warnings": [
    {
      "topic": "testing-renovate-private-dependency",
      "message": "Failed to look up dependency testing-renovate-private-dependency"
    }
  ],
  "versioning": "npm"
}
```

I tried reinstalling the dependency in lower case to make it lower case in package-lock.json as well.

I have a host rule for npm package dependencies from github. Do I need a hostrule to github domain for accessing a private repo as well? Like this: 
```
{
  "extends": [
    "config:base"
  ],
  "hostRules": [
    {
      "matchHost": "https://npm.pkg.github.com/",
      "hostType": "npm",
      "encrypted": {
        "token": "*token-read-package*"
      }
    },
    {
      "matchHost": "github.com",
      "encrypted": {
        "token": "*token-read-repo?*"
      }
    }
  ]
}
```

In that case, what accesses does Renovate need to lookup repo dependencies? 
![image](https://user-images.githubusercontent.com/39908620/143448168-46b51a85-c7a8-4918-a6e2-d2e384a56cd8.png)

Initially I thought Renovate had access to repos and packages on my account since I use the Renovate github app.

**Reply by rarkins** [reply]
Private to private works if both have renovate installed but public to private does not. Try to test with a public repo example with upper case to rule that out

**Reply by Simon-HL** [reply]
Ah, I didn't know Renovate needed to be installed on both repos. The docs about [private packages](https://docs.renovatebot.com/getting-started/private-packages/) says otherwise, from how I understand them: 
_- "Renovate already have credentials for the platform it is running on, i.e. Github, Gitlab etc. If that token has sufficient permissions, this will enable Renovate to lookup **dependencies located in other repositories of the host** or other embedded package registry on the host."_

This is private to private, so it works by also adding Renovate to the repo used as dependency. Thanks!

Can this alternatively be solved by adding a hostRule for github with a PAT for accessing repos if I don't want/can enable Renovate in every repo in an organization?

**Reply by rarkins** [reply]
I disagree with your interpretation of the docs. It says pretty clearly "if the token has sufficient permissions". Even if the statement is confusing, it should be obvious that Renovate has no means to bypass GitHub's access control mechanisms and read from a private repo you have not granted access to.

It's possible that a host rule with a PAT plus matchHost like https://api.github.com/repos/theorg/therepo might work, but I haven't tested it.
